ably/params.go

```diff
@@ -0,0 +1,137 @@
+package ably
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type ProtocolType string
+
+const (
+	ProtocolJSON    = "json"
+	ProtocolMsgPack = "msgpack"
+)
+
+type Params struct {
+	RealtimeEndpoint string
+	RestEndpoint     string
+
+	ApiKey       string
```

Comment:
Ah, OK, well I'll create a new issue to address all of these changes + others from https://github.com/ably/ably-js/issues/33